### PR TITLE
feat: card pagination

### PR DESCRIPTION
**Pull request by shawnyama**
# Description
- Add pagination indicator / nav control to the carousel
- Made carousel its own component so indicators and arrow scrolling can be reused

https://github.com/DARPA-ASKEM/Terarium/assets/28237258/ac8b86ec-9c28-48c5-bb0f-61ff6cc38f52

Sorry about all the commits I kept branching off my own stuff and later merging it back to main

### CODE REVIEW

**Comment by shawnyama** [comment]
Found an edge case where if you are in the middle of the carousel and then resize your window the page position gets messed up. Let me know if I should save fixing that for a separate PR or if I should fix it here.

File: packages/client/hmi-client/src/components/home/tera-card-carousel.vue [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+const carouselRef = ref();
+const cardListRef = ref();
+const carouselWidth = ref(0);
+const activeCarouselPage = ref(1);
+
+const amountOfCardsToMove = computed(() =>
+	carouselWidth.value ? Math.floor(carouselWidth.value / (cardWidth + rightGapWidth)) : 3
+);
+
+const scrollIncriment = computed(() => (cardWidth + rightGapWidth) * amountOfCardsToMove.value);
```
Comment (echl) [inline packages/client/hmi-client/src/components/home/tera-card-carousel.vue:69]: typo - increment

File: packages/client/hmi-client/src/components/home/tera-card-carousel.vue [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+const cardWidth = 238; // 17rem
+const rightGapWidth = 21; // 1.5rem
+
+const carouselRef = ref();
+const cardListRef = ref();
+const carouselWidth = ref(0);
+const activeCarouselPage = ref(1);
+
+const amountOfCardsToMove = computed(() =>
+	carouselWidth.value ? Math.floor(carouselWidth.value / (cardWidth + rightGapWidth)) : 3
```
Comment (echl) [inline packages/client/hmi-client/src/components/home/tera-card-carousel.vue:66]: Looks like `carouselWidth` will always have a value, so no need for the conditional?

File: packages/client/hmi-client/src/components/home/tera-card-carousel.vue [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+				bounds &&
+				carouselBounds &&
+				bounds.x + bounds.width < carouselBounds.x + carouselWidth.value
+			) {
+				return;
+			}
+		}
+		activeCarouselPage.value += pagesToMove;
+		pixelsToTranslate = -scrollIncriment.value * pagesToMove;
+	} else if (direction === 'left') {
```
Comment (echl) [inline packages/client/hmi-client/src/components/home/tera-card-carousel.vue:101]: Looks like could get rid of `direction` entirely by setting `pagesToMove` to be a negative/positive number depending on if the left/right scroll arrow is clicked.

File: packages/client/hmi-client/src/components/home/tera-card-carousel.vue [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+		cardListRef.value.style.marginLeft === '' ? '0' : cardListRef.value.style.marginLeft;
+	const currentMarginLeft = parseFloat(marginLeftString);
+	const newMarginLeft = currentMarginLeft + pixelsToTranslate;
+	// Don't let the list scroll far enough left that we see space before the first card.
+	cardListRef.value.style.marginLeft = `${newMarginLeft > 0 ? 0 : newMarginLeft}px`;
+}
+
+function onCarouselPaginate(page: number) {
+	if (page === activeCarouselPage.value) return;
+	const pagesToMove = Math.abs(activeCarouselPage.value - page);
```
Comment (echl) [inline packages/client/hmi-client/src/components/home/tera-card-carousel.vue:116]: Echoing the comment about getting rid of `direction` in the `scroll` function, if you change this to `const pagesToMove = page - activeCarouselPage.value`, then the sign of `pagesToMove` will reflect which direction the carousel should scroll.

**Review by YohannParis — APPROVED** [review]

**Pull request merged**